src/oas/operation.ts

```diff
@@ -57,7 +57,16 @@ export const transformOasOperation: TranslateFunction<
 
   const serviceId = (this.ids.service = String(this.document['x-stoplight']?.id));
   this.ids.path = this.generateId.httpPath({ parentId: serviceId, path });
```

Comment:
```suggestion
  const serviceId = (this.ids.service ||= String(this.document['x-stoplight']?.id));
  this.ids.path ||= this.generateId.httpPath({ parentId: serviceId, path });
```
perhaps make it optional so that they don't get re-assigned

Reply:
Changing this resulted in re-assignment not working when it was supposed to (almost all the tests started failing with the incorrect serviceId etc). I will not be making this change as part of the PR - should I open an issue that covers this?

Reply:
ah, I know why the tests fail.
There's no bug in the code, the bug would be actually introduced if you applied my suggestion 😆